Assets/Scripts/CharacterInfo.cs

```diff
@@ -36,42 +31,122 @@ public class CharacterInfo : MonoBehaviour {
 	}
 
 	void Update (){
-		// if (currentEXP >= expChart.expToLevel[characterLevel - 1]){
-		// 	LevelUp();
-		// }
+		/* 	Still don't know if CharacterInfo actually needs an Update method. We can probably
+			make any changes we need to do from other scripts through public methods on this
+			script. Leaving it for now, but we may end up foregoing Update altogether.
+		*/
 	}
 
     public void PutItemInInventory(Item item){
 		inventoryList.Add(item);
 	}
 
+	public void GiveExp(int exp){
+		currentEXP += exp;
+		if (currentEXP >= expChart.expToLevel[characterLevel - 1]){
+			LevelUp();
+		}
+	}
+
 	private void LevelUp(){
+		int oldLevel = characterLevel;
 		characterLevel++;
+		int r, rVitIQ;
 
-		#region Base Stats Increase
-		int tempIQ = Random.Range(1,6);
-		iq += tempIQ;
-		// Text.text = "IQ increased by " + tempIQ "points!";
-		guts += Random.Range(1,6);
-		luck += Random.Range(1,6);
-		vitality += Random.Range(1,6);
-		speed += Random.Range(1,6);
-		offense += Random.Range(1,6);
-		defense += Random.Range(1,6);
-		#endregion
-		
-		if (vitality % 5 == 0){
-			maxHP++;
+		#region Character Specific Stat Growth Rates
```

Comment:
I would say, if you need to region a block of code in a method, it probably needs to be extracted into its own method